### PR TITLE
feat: switch pagetype to insertionstart

### PR DESCRIPTION
**Pull request by danbosnichill**
BREAKING CHANGE: This does a breaking change to the DeliveryRequest interface.

Motivation: support sending request insertions in blocks.

TESTING=unit tests

### CODE REVIEW

File: src/delivery-request.ts [hunk 1]
```diff
@@ -11,6 +10,30 @@ export interface DeliveryRequest {
...
+   * a. If there are 10 items and all 10 items are in `request.insertion`, then
+   *    insertionStart=0.
+   * b. If there are 10,000 items and the first 500 items are on `request.insertion`,
+   *    then insertionStart=0.
+   * c. If there are 10,000 items and we want to send items [500,1000) on `request.insertion`,
+   *    then insertionStart=500.
+   * c. If there are 10,000 items and we want to send items the last page [9500,10000) on `request.insertion`,
+   *    then insertionStart=9500.
+   *
+   * This field is required because it will likely be a source of bugs.
```
Comment (scottmcmaster) [inline src/delivery-request.ts:31]: ?

**Review by scottmcmaster — APPROVED** [review]
If this isn't WIP anymore, does it need a different title for semver for the nature of the change?

File: src/delivery-request.ts (hunk 1 above)
Comment (danbosnichill) [reply src/delivery-request.ts:31]: Sorry, fixing the comment now.

File: src/pager.ts [hunk 2]
```diff
@@ -1,37 +1,36 @@
-import { InsertionPageType } from './insertion-page-type';
 import { Insertion, Paging } from './types/delivery';
 
+export const getOffset = (paging: Paging | undefined) => {
+  const offset = paging?.offset ?? 0;
```
Comment (scottmcmaster) [inline src/pager.ts:4]: It seems like maybe you could combine these lines (but not essential)

**Review by scottmcmaster — APPROVED** [review]

File: src/pager.ts (hunk 2 above)
Comment (danbosnichill) [reply src/pager.ts]: Done.

**Pull request merged**